### DISCUSSION TITLE
How can I use local (from my computer) images in my userscripts?

### DISCUSSION THREAD

**Discussion by atheist1917** [violentmonkey/violentmonkey#1534] [Q&A]
Because protocol `file://` is prohibited.

**Comment by tophf** [comment]
It's best to run any localhost web server and use an `http://localhost/foo/bar.jpg` URL.
A less secure alternative (not recommended) is to enable file access on `chrome://extensions` page in details of Violentmonkey.

**Comment by atheist1917** [comment]
I cant open `chrome://extensions` page

**Comment by tophf** [comment]
You probably have a malware extension that blocks opening this page. Try right-clicking Violentmonkey icon in the toolbar and choose "Manage extension".

**Comment by atheist1917** [comment]
Are you sure I can open `chrome://extensions` in Firefox?

**Comment by tophf** [comment]
In Firefox extensions can't open file://

**Comment by atheist1917** [comment]
OK. How can I make `blob:` from local image to insert as `src`?

**Reply by gera2ld** [reply]
Maybe you can embed the image into the script as base64, you shouldn't use a local file directly in JavaScript anyway.
It's just how JavaScript works, you can learn more on StackOverflow.

**Reply by atheist1917** [reply]
It's small SVG images, not JPG or PNG, so it's small. How can I make blob from svg image file?

**Reply by tophf** [reply]
You can use an online converter from svg to data uri like [this one](https://yoksel.github.io/url-encoder/): paste svg on the left and use the result at the bottom starting with `data:image/svg+xml,` until the double quote at the end, not including it.

**Reply by gera2ld** [reply]
FYI, there are many examples on Stack Overflow
- https://stackoverflow.com/search?q=inline+svg
- https://stackoverflow.com/search?q=svg+blob
